### ISSUE TITLE
netbox does not read some error messages

### ISSUE THREAD

**Issue by Gerold103** [tarantool/tarantool#3874]
**Tarantool version:**
Any

**OS version:**
Any

**Bug description:**
If a connection attempt has failed due with a ClientError, this error is not read by netbox. This is because of how error sending works. A server, getting ClientError on connect, sends back a greeting string (128 bytes), an iproto packet with error message and closes the socket. Netbox reads greeting and then tries to fetch schema. For this netbox tries to send a request to a server. Of course, it gets an error that the connection is closed already. In such a case netbox finishes worker fiber, frees resources etc. But there were some data to read. If netbox had tried to read after failed write, it could have got the actual error.

**Steps to reproduce:**
```Lua
box.cfg{listen = 3313}
box.schema.user.grant('guest', 'read,write,execute', 'universe')
netbox = require('net.box')
c = netbox.connect(box.cfg.listen)
_ = c._transport.perform_request(nil, nil, 'inject', nil, nil, '\x80')
function break_connection()
	error('Connections are not allowed')
end
box.session.on_connect(break_connection)
c = netbox.connect(box.cfg.listen)
```
Here error message `'Connections are not allowed'` is sent, but not read. Netbox reports error 'Peer closed', not the actual message.

**Comment by kyukhin** [comment]
We need to reproduce this.

**Comment by sergos** [comment]
It needs some update to the inject function call due to (multiple) changes of this internal interface
```
_ = c._transport:perform_request(0, nil, nil, nil, nil, nil, nil, nil, 'inject', '\x80', nil, nil, nil, nil)
```

The problem is fixed with c13b3a31a
